### PR TITLE
Enable parallel option in Delta Lake product tests

### PR DESCRIPTION
**Pull request by ebyhr**
## Description

Delta Product tests sometimes take 2 hours in total.
Relates to #18805

## Release notes

(x) This is not user-visible or is docs only, and no release notes are required.

### CODE REVIEW

**Review by wendigo — APPROVED** [review]

**Comment by wendigo** [comment]
Does it work as expected? How can I verify that test indeed was executed in parallel?

File: pom.xml [hunk 1]
```diff
@@ -164,7 +164,7 @@
         <dep.jsonwebtoken.version>0.11.5</dep.jsonwebtoken.version>
         <dep.oracle.version>21.9.0.0</dep.oracle.version>
         <dep.drift.version>1.21</dep.drift.version>
-        <dep.tempto.version>200</dep.tempto.version>
+        <dep.tempto.version>201</dep.tempto.version>
```
Comment (findinpath) [inline pom.xml:167]: Notes for other reviewers: relates to https://github.com/trinodb/tempto/pull/122

**Comment by ebyhr** [comment]
>  How can I verify that test indeed was executed in parallel?

The log indicates tests ran in parallel. However, `suite-delta-lake-oss` faced concurrency issues. I will investigate the cause.

**Comment by ebyhr** [comment]
Closing as we're trying a different approach.

**Pull request closed without merging**